Framework/SINQ/inc/MantidSINQ/SliceMDHisto.h

```diff
@@ -43,7 +43,7 @@ class MANTID_SINQ_DLL SliceMDHisto : public Mantid::API::Algorithm {
   unsigned int m_rank;
   std::vector<int> m_dim;
   void cutData(const Mantid::API::IMDHistoWorkspace_sptr &inWS, const Mantid::API::IMDHistoWorkspace_sptr &outWS,
-               Mantid::coord_t *sourceDim, Mantid::coord_t *targetDim, std::vector<int> start, std::vector<int> end,
+               Mantid::coord_t *sourceDim, Mantid::coord_t *targetDim, std::vector<int> &start, std::vector<int> &end,
```

Comment:
```suggestion
               Mantid::coord_t *sourceDim, Mantid::coord_t *targetDim, const std::vector<int> &start, const std::vector<int> &end,
```
These should be passed by `const` ref, otherwise there's a possibility that the values in the parent function could be changed.